spec/mail/notify/mailer_spec.rb

```diff
@@ -1,53 +1,219 @@
 
-  context "with a view" do
-    it "sends the template" do
-      expect(mailer).to receive(:mail).with({template_id: "foo", bar: "baz"})
-      mailer.view_mail("foo", bar: "baz")
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.template_id).to eql("template-id")
+    end
+
+    it "sets the message subject" do
+      message_params = {template_id: "template-id", to: "test.name@email.co.uk", subject: "Test subject"}
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.header[:subject]).to be_a Mail::Field
+      expect(message.header[:subject].value).to eql("Test subject")
     end
 
-    it "raises an error if the template ID is blank" do
-      expect { mailer.view_mail("", bar: "baz") }.to raise_error(
-        ArgumentError,
-        "You must specify a template ID"
+    it "sets the message to address" do
+      message_params = {template_id: "template-id", to: "test.name@email.co.uk", subject: "Test subject"}
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.header[:to]).to be_a Mail::Field
+      expect(message.header[:to].value).to eql("test.name@email.co.uk")
+    end
+
+    it "sets the subject on personalisation" do
+      message_params = {template_id: "template-id", to: "test.name@email.co.uk", subject: "Test subject"}
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.personalisation[:subject]).to eql("Test subject")
+    end
+
+    it "sets the body on personalisation" do
+      message_params = {template_id: "template-id", to: "test.name@email.co.uk", subject: "Test subject"}
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.personalisation[:body]).to eql("This is the body from the mailers view.\r\n")
+    end
+
+    it "drops any further personalisation" do
+      message_params = {
+        template_id: "template-id",
+        to: "test.name@email.co.uk",
+        subject: "Test subject",
+        personalisation: {name: "Test Name"}
+      }
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.personalisation.key?(:name)).to be false
+    end
+
+    it "sets the reply_to_id" do
+      message_params = {
+        template_id: "template-id",
+        to: "test.name@email.co.uk",
+        subject: "Test subject",
+        reply_to_id: "test-reply-to-id"
+      }
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.reply_to_id).to eql("test-reply-to-id")
+    end
+
+    it "sets the reference" do
+      message_params = {
+        template_id: "template-id",
+        to: "test.name@email.co.uk",
+        subject: "Test subject",
+        reply_to_id: "test-reference"
+      }
+
+      message = TestMailer.with(message_params).test_view_mail
+
+      expect(message.reply_to_id).to eql("test-reference")
+    end
+
+    it "requires template id to be set" do
```

Comment:
Interesting, feels like taste again rather than better or worse!  

I think I was after that AAA goodness here, I think I have been burned by over use of `context` and `let` in the past and so tend to avoid them, but I can see why you like the approach.

At this stage I am happy to keep it as is as long as we have coverage.